modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/SqlListenerUtils.java

```diff
@@ -134,7 +134,7 @@
                 reader.in().position(reader.in().position() - 1);
 
                 if (binObjAllow)
-                    return reader.readObjectDetached();
+                    return GridBinaryMarshaller.KEEP_BINARIES.get() ? reader.readObjectDetached() : reader.readObject();
```

Comment:
Is value of `GridBinaryMarshaller.KEEP_BINARIES` set somewhere?

Reply:
Yes, it is set to `connProps.isKeepBinary()` just before sending request. See [this](https://github.com/gridgain/gridgain/pull/307/files/eebf76e0ec1799b094deed6d5067029ab9ffa17b#diff-03693a98b4b30190856ac0f42d9b3d7fR993)